src/wallet/wallet.h

```diff
@@ -274,6 +274,15 @@ class CReserveKey
     //! Destructor. If a key has been reserved and not KeepKey'ed, it will be returned to the keypool
     ~CReserveKey()
     {
+#ifdef DEBUG
+        // To avoid accidental key re-use, we should attempt to have the creator
+        // have it returned or kept before its destructor is called.
+        // TODO: Fix the lifecycle of this object to be less error-prone!
+        assert(nIndex == -1);
+#endif
+        if (nIndex != -1) {
+            LogPrintf("Warning! A CReserveKey was not dealt with until destructor. This is a bug. Please report this if possible.");
```

Comment:
What about.

```suggestion
            LogPrintf("Warning! A CReserveKey was not dealt with until destructor. This is a bug. Please report this if possible, e.g. here: " PACKAGE_BUGREPORT "/new?title=wallet: ReserveKey was not dealt with until destructor");
```

Reply:
wrestled with tinyformat a bit, but took heart of suggestion

Reply:
```
2019-06-13T14:45:00Z Error "tinyformat: Too many conversion specifiers in format string" while formatting log message: Warning! A CReserveKey was not dealt with until destructor. This is a bug. Please report this if possible , e.g. here: https://github.com/bitcoin/bitcoin/issues/new?title=wallet:%20ReserveKey%20was%20not%20dealt%20with%20until%20destructor
```
to
```
2019-06-13T14:50:09Z Warning! A CReserveKey was not dealt with until destructor. This is a bug. Please report this if possible , e.g. here: https://github.com/bitcoin/bitcoin/issues/new?title=wallet:%20ReserveKey%20was%20not%20dealt%20with%20until%20destructor
```

Was complaining about the `%` so had to double-up those so the checker wouldn't complain.